File: fixtures/java/Animal.cs
using System;

namespace Example.Models
{
    /// <summary>
    /// 动物基类
    /// </summary>
    public class Animal
    {
        protected string name;
        protected int age;
        
        public Animal(string name, int age)
        {
            this.name = name;
            this.age = age;
        }
        
        public virtual void MakeSound()
        {
            Console.WriteLine("Some generic animal sound");
        }
        
        public virtual void Eat(string food)
        {
            Console.WriteLine($"{name} is eating {food}");
        }
        
        public string GetName() => name;
        public int GetAge() => age;
    }
}

File: fixtures/java/Animal.java
package com.example.model;

/**
 * 动物基类
 */
public class Animal {
    protected String name;
    protected int age;
    
    public Animal(String name, int age) {
        this.name = name;
        this.age = age;
    }
    
    public void makeSound() {
        System.out.println("Some generic animal sound");
    }
    
    public void eat(String food) {
        System.out.println(name + " is eating " + food);
    }
    
    public String getName() {
        return name;
    }
    
    public int getAge() {
        return age;
    }
}
